### PR TITLE
[jenkins] add mac build script

### PR DESCRIPTION
**Pull request by heliclei**

### CODE REVIEW

File: tools/jenkins-scripts/mac-build.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+COCOS2DX_ROOT="$DIR"/../..
+xcodebuild -project "$COCOS2DX_ROOT"/build/cocos2d_samples.xcodeproj -scheme "build all tests iOS"  -destination "platform=OS X" clean
+xcodebuild -project "$COCOS2DX_ROOT"/build/cocos2d_samples.xcodeproj -scheme "build all tests iOS"  -destination "platform=OS X" build
```
Comment (dumganhar) [inline tools/jenkins-scripts/mac-build.sh]: The scheme should be `build all samples Mac`

Comment (heliclei) [inline tools/jenkins-scripts/mac-build.sh]: Oops, such a Mindless bug!

File: tools/jenkins-scripts/mac-build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+COCOS2DX_ROOT="$DIR"/../..
+xcodebuild -project "$COCOS2DX_ROOT"/build/cocos2d_samples.xcodeproj -scheme "build all samples Mac"  -destination "platform=OS X" clean
+xcodebuild -project "$COCOS2DX_ROOT"/build/cocos2d_samples.xcodeproj -scheme "build all samples Mac"  -destination "platform=OS X" build
```
Comment (dumganhar) [inline tools/jenkins-scripts/mac-build.sh:4]: Ouch, we have reverted `cocos2d_samples` to `cocos2d_tests`, `build all samples Mac` to `build all tests Mac`

**Comment by cocosQA** [comment]
Dear heliclei:
  We have created a new branch 'v3' to replace branch 'develop', and this PR has been moved from 'develop' branch to 'v3' branch, the new PR is https://github.com/cocos2d/cocos2d-x/pull/6449
The old 'develop' branch will be deleted soon, and this PR will be closed accordingly.

**Pull request closed without merging**